app/Filament/Resources/DocumentResource/RelationManagers/RevisionsRelationManager.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Filament\Resources\DocumentResource\RelationManagers;
+
+use App\Filament\Resources\RevisionResource;
+use App\Models\Document;
+use App\Models\Revision;
+use Filament\Forms;
+use Filament\Forms\Form;
+use Filament\Resources\RelationManagers\RelationManager;
+use Filament\Tables;
+use Filament\Tables\Table;
+use Illuminate\Support\Carbon;
+use Livewire\Features\SupportFileUploads\TemporaryUploadedFile;
+
+class RevisionsRelationManager extends RelationManager
+{
+    protected static string $relationship = 'revisions';
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Forms\Components\DatePicker::make('date')
+                    ->label(__('Revision date'))
+                    ->format('Y-m-d')
+                    ->unique()
+                    ->default(Carbon::now())
+                    ->validationMessages([
+                        'unique' => __('A revision with this date already exists.'),
+                    ])
+                    ->columnSpan(2),
+                Forms\Components\FileUpload::make('file.en')
+                    ->label(__('File (English)'))
+                    ->requiredWithout('file.fr')
+                    ->disk('public')
+                    ->directory('documents')
+                    ->getUploadedFileNameForStorageUsing(function (?Revision $record, TemporaryUploadedFile $file, RelationManager $livewire): string {
```

Comment:
`$record` is `null`. When it gets passed to `getFilename` it then defaults to the current date, so all of the files are still saved with the current date.

Reply:
I haven't yet been able to find a way to access the record from inside this function when called via the relation manager.